src/groups.jl

```diff
@@ -364,7 +364,11 @@ function benchmarkset_block(title, ex::Expr)
     stack = GlobalRef(BenchmarkTools, :benchmark_stack)
     quote
         let $(Symbol("#root#")) = last($stack)
-            $(Symbol("#root#"))[$title] = $(Symbol("#suite#")) = BenchmarkGroup()
+            if !haskey($(Symbol("#root#")), $title)
+                $(Symbol("#root#"))[$title] = $(Symbol("#suite#")) = BenchmarkGroup()
+            else
+                $(Symbol("#suite#")) = $(Symbol("#root#"))[$title]
```

Comment:
Can you explain to me why `$(Symbol("#suite#"))` is set based on `$(Symbol("#root#"))[$title]` here, but two lines above it's the reverse? I'm just discovering this code base

Reply:
`#suite#` should refer to the BenchmarkGroup `$title` under `#root#`. However, that group might not exist (yet) and so we check if it does, and if not initiate it.

The problem with the original code was, that it only had
```
$(Symbol("#root#"))[$title] = $(Symbol("#suite#")) = BenchmarkGroup()
```
and thus the BenchmarkGroup `$title` was overwritten at every loop iteration.


Here is the generated code from the test case (formatted with `prettify` from `MacroTools`)
```
quote
    if isempty(BenchmarkTools.benchmark_stack)
        push!(BenchmarkTools.benchmark_stack, (BenchmarkGroup)())
    end
    for k = 1:2
        let goat = last(BenchmarkTools.benchmark_stack)
            if !(haskey(goat, "test set"))
                goat["test set"] = (hummingbird = BenchmarkGroup())
            else
                hummingbird = goat["test set"]
            end
            push!(BenchmarkTools.benchmark_stack, hummingbird)
            hummingbird["test case $(k)"] = BenchmarkTools.generate_benchmark_definition(Main, Symbol[], Any[], [Symbol("##k#316"), Symbol("##k#317")], (k, k), $(Expr(:copyast, :($(QuoteNode(:(var"##k#316" + var"##k#317")))))), $(Expr(:copyast, :($(QuoteNode(nothing))))), $(Expr(:copyast, :($(QuoteNode(nothing))))), BenchmarkTools.Parameters())
            pop!(BenchmarkTools.benchmark_stack)
        end
    end
    if length(BenchmarkTools.benchmark_stack) == 1
        pop!(BenchmarkTools.benchmark_stack)
    end
end
```

Reply:
Does this get easier due to the recently-merged #309 ?

Reply:
It probably does! I hadn't seen that PR. I'll check it out.

Reply:
Yeah, I just got ~~hired~~ trapped as a maintainer so I'm cleaning up the house ^^

Reply:
Thanks for taking on the responsibility!

I've adapted the code according to #309. Very convenient.

Reply:
OK I don't fully understand what's going on but the test seems to work and the code change is now one line so I'm choosing trust